storybook/qmlTests/TestComponents/tst_StatusChatInput.qml

```diff
@@ -1,21 +1,571 @@
+                compare(controlUnderTest.textInput.length, standardInput.length, "Expected identical text length")
+                compare(controlUnderTest.textInput.getText(0, controlUnderTest.textInput.length),
+                        standardInput.getText(0, standardInput.length),
+                        "Expected identical text")
+                compare(controlUnderTestCursorPosition, standardControlCursorPosition, "Expected identical cursor position")
+                compare(controlUnderTestSelection, standardControlSelection, "Expected identical selected text")
+            } else {
+                //the text is expected to be different due to custom processor
+                //Ex: bold or italic where text is wrapped in specific symbols
+                verify(controlUnderTest.textInput.getText(0, controlUnderTest.textInput.length) !==
+                       standardInput.getText(0, standardInput.length),
+                       "Expected different text")
+            }
+        }
+    }
+
+    TestCase {
+        id: statusChatInputEmojiAndMentions
+        name: "StatusChatInputEmojiAndMentions"
+        when: windowShown
+
+        property StatusChatInput controlUnderTest: null
+
+        function init() {
+            Utils.globalUtilsInst = globalUtilsMock
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+        }
+
+//        Scenario: The user can type text, mention and emoji
+//        Given a new instance of StatusChatInput
+//        And the user has  <mention> as contact
+//        When the user is typing <text>
+//        Then the text is displayed in the input field as <expectedText>
+//        And the <mention> is inserted in the input field
+//        And the <emoji> is inserted in the input field
+//        And the <mention> can be replaced with contact public key
+
+//        Examples:
+//        text                              | mention              	| expectedPlainText
+//        “Hello John:D”                    |                       | “Hello John\uD83D\uDE04“
+//        “Hello @JohnDoe”                  | @JohnDoe				| “Hello @JohnDoe ”
+//        “Hello:D@JohnDoe“                 | @JohnDoe				| “Hello\uD83D\uDE04 @JohnDoe “
+//        “:DHello@JohnDoe”                 | @JohnDoe 				| “\uD83D\uDE04 Hello @JohnDoe ”
+//        “:D:D:D:D:D:D@JohnDoe:D:D:D”		| @JohnDoe				| “\uD83D\uDE04 \uD83D\uDE04 \uD83D\uDE04 \uD83D\uDE04 \uD83D\uDE04 \uD83D\uDE04 @JohnDoe \uD83D\uDE04 \uD83D\uDE04 \uD83D\uDE04 ”
+//        “Hello:@JohnDoe1:D@JohnDoe2:D 	| @JohnDoe1 @JohnDoe2	| “Hello:@JohnDoe1 \uD83D\uDE04 @JohnDoe2 \uD83D\uDE04 ”
+//        “Hello:@JohnDoe1:D@JohnDoe2:D 	|                       | “Hello:@JohnDoe1\uD83D\uDE04 @JohnDoe2\uD83D\uDE04 ”
+        function test_text_mention_emoji_input_data() {
+            return [
+                { tag: "Hello John:D", text: "Hello John:D", mention: [], expectedText: "Hello John\uD83D\uDE04 " },
```

Comment:
Suggestion: Following the same comment line than the global data object, maybe it makes sense for specific test cases, to create a specific data object to define the specific needed data, to have a cleaner code, like in this example.